### PR TITLE
Adds the Maintenance Jack

### PR DESCRIPTION
**Pull request by BeagleGaming1**
# About the pull request

This PR gives the Maintenance Jack to ship synths in the experimental vendor and Working Joes for free.
Gives the Maintenance Jack bonus effects
(Had to give WJs Engi 3 for one of the Maintenance Jack's abilities, this shouldn't change much)

# Explain why it's good for the game

New tool for synths and WJs

# Changelog

:cl:
add: Added a Maintenance Jack for Synthetics (Experimental Vendor) and Working Joes (Standard Kit). It has two modes, being able to switch between a crowbar and a wrench. You can use it to unbolt doors, and, if strong enough, pry them open.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -279,3 +279,116 @@
 					M.apply_damage(power,BURN,def_zone)
 
 	return 1
+
+/obj/item/weapon/melee/maintenance_jack //put here for unique action
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:283]: why is this `/obj/item/weapon/melee`

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -606,6 +606,25 @@ GLOBAL_LIST_INIT(airlock_wire_descriptions, list(
 				update_icon()
 		return
 
+	if(istype(C, /obj/item/weapon/melee/maintenance_jack) && locked)
+		var/obj/item/weapon/melee/maintenance_jack/current_jack = C
+		if(!skillcheck(user, SKILL_ENGINEER, SKILL_ENGINEER_MASTER)) //Only Engi 3 can use that part of the ability
+			to_chat(user, SPAN_NOTICE("You can't figure out how to disable the bolts."))
+			return FALSE
+		if(current_jack.mode == "wrench")
```
Comment (Zonespace27) [inline code/game/machinery/doors/airlock.dm:614]: have the string be a define or something

File: code/game/machinery/doors/airlock.dm [hunk 3]
```diff
@@ -606,6 +606,25 @@ GLOBAL_LIST_INIT(airlock_wire_descriptions, list(
...
+			user.visible_message(SPAN_DANGER("[user] begins to disable [src]'s bolts!"),\
+			SPAN_NOTICE("You start to disable [src]'s bolts."))
+			playsound(src, "pry", 25, TRUE)
+
+			if(!do_after(user, 5 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE, src, INTERRUPT_ALL))
+				to_chat(user, SPAN_WARNING("You decide not to disable the bolts on [src] airlock."))
+				return
+
+			user.visible_message(SPAN_DANGER("[user] disables the bolts on [src] airlock."),\
+			SPAN_NOTICE("You unbolt [src] airlock."))
```
Comment (Zonespace27) [inline code/game/machinery/doors/airlock.dm:624]: how does this look in-game, considering the `[src] airlock` part

File: code/game/objects/items/weapons/weaponry.dm [hunk 4]
```diff
@@ -279,3 +279,116 @@
...
+	name = "K92 Maintenance Jack"
+	desc = "A combination crowbar, wrench, and generally large bludgeoning device that comes in handy in emergencies. Can be used to disengage door jacks. Pretty hefty, though."
+	icon_state = "maintenance_jack"
+	item_state = "maintenance_jack"
+	hitsound = "swing_hit"
+	w_class = SIZE_LARGE
+	force = MELEE_FORCE_STRONG
+	inherent_traits = list(TRAIT_TOOL_CROWBAR)
+	var/mode = "crowbar"
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:292]: code doc + put at bottom of type declaration

File: code/game/objects/items/weapons/weaponry.dm [hunk 5]
```diff
@@ -279,3 +279,116 @@
...
+	flags_equip_slot = SLOT_SUIT_STORE
+	pry_capable = IS_PRY_CAPABLE_FORCE //but not really
+
+/obj/item/weapon/melee/maintenance_jack/verb/use_unique_action()
+	set category = "Weapons"
+	set name = "Flip Maintenance Jack"
+	set desc = "Toggle the mode of the maintenance jack."
+	set src in usr
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:300]: this needs to be an actionbutton or use-in-hand functionality as well

File: code/game/objects/items/weapons/weaponry.dm [hunk 6]
```diff
@@ -279,3 +279,116 @@
...
+	unique_action(usr)
+
+/obj/item/weapon/melee/maintenance_jack/get_examine_text(mob/user)
+	. = ..()
+	. += SPAN_NOTICE("Use the Flip action to switch modes.")
+	if(mode == "crowbar")
+		. += SPAN_NOTICE("It is set to crowbar mode, allowing you to pry open doors, provided you are strong enough.")
+	if(mode == "wrench")
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:308]: else if?

File: code/game/objects/items/weapons/weaponry.dm [hunk 7]
```diff
@@ -279,3 +279,116 @@
...
+		. += SPAN_NOTICE("It is set to wrench mode, allowing you to unbolt doors, provided you are smart enough to know how.")
+
+/obj/item/weapon/melee/maintenance_jack/unique_action(mob/living/user)
+	if(user.get_active_hand() != src)
+		return
+
+	playsound(src, 'sound/weapons/punchmiss.ogg', 15, TRUE, 3)
+	if(mode == "crowbar") //Wrench mode | Remove bolts if engi 3
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:316]: use defines, as said before

File: code/game/objects/items/weapons/weaponry.dm [hunk 8]
```diff
@@ -279,3 +279,116 @@
...
+		mode = "crowbar"
+		animate(src, transform = matrix(180, MATRIX_ROTATE), time = 2, easing = EASE_IN)
+		animate(transform = matrix(270, MATRIX_ROTATE), time = 1)
+		animate(transform = matrix(360, MATRIX_ROTATE), time = 2, easing = EASE_OUT)
+		REMOVE_TRAIT(src, TRAIT_TOOL_WRENCH, TRAIT_SOURCE_INHERENT)
+		ADD_TRAIT(src, TRAIT_TOOL_CROWBAR, TRAIT_SOURCE_INHERENT)
+		pry_capable = IS_PRY_CAPABLE_FORCE
+
+/obj/item/weapon/melee/maintenance_jack/afterattack(obj, mob/living/user, proximity)
+	if(!(mode == "crowbar")) //Otherwise it lets you pry open right after unbolting
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:339]: `!=` exists

File: code/game/objects/items/weapons/weaponry.dm [hunk 9]
```diff
@@ -279,3 +279,116 @@
...
+		return
+	if(!(HAS_TRAIT(user, TRAIT_SUPER_STRONG))) //Only synths can force open doors, otherwise its a regular crowbar
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:341]: extra set of brackets unneeded

File: code/game/objects/items/weapons/weaponry.dm [hunk 10]
```diff
@@ -279,3 +279,116 @@
...
+		SPAN_NOTICE("You change your grip on [src]. You will now use it as a crowbar."))
+		mode = "crowbar"
+		animate(src, transform = matrix(180, MATRIX_ROTATE), time = 2, easing = EASE_IN)
+		animate(transform = matrix(270, MATRIX_ROTATE), time = 1)
+		animate(transform = matrix(360, MATRIX_ROTATE), time = 2, easing = EASE_OUT)
+		REMOVE_TRAIT(src, TRAIT_TOOL_WRENCH, TRAIT_SOURCE_INHERENT)
+		ADD_TRAIT(src, TRAIT_TOOL_CROWBAR, TRAIT_SOURCE_INHERENT)
+		pry_capable = IS_PRY_CAPABLE_FORCE
+
+/obj/item/weapon/melee/maintenance_jack/afterattack(obj, mob/living/user, proximity)
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:338]: why is `obj` not typecasted
also change the var name

File: code/game/objects/items/weapons/weaponry.dm [hunk 11]
```diff
@@ -279,3 +279,116 @@
...
+	if(!(mode == "crowbar")) //Otherwise it lets you pry open right after unbolting
+		return
+	if(!(HAS_TRAIT(user, TRAIT_SUPER_STRONG))) //Only synths can force open doors, otherwise its a regular crowbar
+		to_chat(user, SPAN_DANGER("The airlock's motors resist your efforts to force it."))
+		return
+	if(get_dist(src, obj) > 1)
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:344]: `proximity` arg?

File: code/game/objects/items/weapons/weaponry.dm [hunk 12]
```diff
@@ -279,3 +279,116 @@
...
+		return
+
+	if(istype(obj, /obj/structure/machinery/door/airlock))
+		var/obj/structure/machinery/door/airlock/door = obj
+		if(!(HAS_TRAIT(user, TRAIT_SUPER_STRONG))) //basically IS_PRY_CAPABLE_CROWBAR
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:349]: second set of brackets not needed

File: code/game/objects/items/weapons/weaponry.dm [hunk 13]
```diff
@@ -279,3 +279,116 @@
...
+			if(!door.arePowerSystemsOn())
+				if(!door.operating)
+					if(door.density)
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:352]: is there really not a var for if the door is open or not

File: code/game/objects/items/weapons/weaponry.dm [hunk 14]
```diff
@@ -279,3 +279,116 @@
...
+		if(door.heavy)
+			to_chat(usr, SPAN_DANGER("You cannot force [door] open."))
+			return FALSE
+		if(user.action_busy || user.a_intent == INTENT_HARM)
+			return
+
+		user.visible_message(SPAN_DANGER("[user] jams [src] into [door] and starts to pry it open."),
+		SPAN_DANGER("You jam [src] into [door] and start to pry it open."))
+		playsound(src, "pry", 15, TRUE)
+		if(do_after(user, 3 SECONDS, INTERRUPT_ALL, BUSY_ICON_HOSTILE))
```
Comment (Zonespace27) [inline code/game/objects/items/weapons/weaponry.dm:370]: early return

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by carlarctg** [comment]
improve your changelog it doenst say anything

**Review by Agameofscones — CHANGES_REQUESTED** [review]
These inhands are comically oversized. Consider designing around a modified crowbar inhand for a better scale.

**Review by nauticall — APPROVED** [review]
alright, good on ya for fixing em up, sprites approved now! 👍

File: code/game/objects/items/tools/maintenance_tools.dm [hunk 15]
```diff
@@ -490,33 +490,116 @@
...
+		user.visible_message(SPAN_INFO("[user] changes their grip on [src]. They will now use it as a crowbar."),
+		SPAN_NOTICE("You change your grip on [src]. You will now use it as a crowbar."))
+		crowbar_mode = TRUE
+		animate(src, transform = matrix(180, MATRIX_ROTATE), time = 2, easing = EASE_IN)
+		animate(transform = matrix(270, MATRIX_ROTATE), time = 1)
+		animate(transform = matrix(360, MATRIX_ROTATE), time = 2, easing = EASE_OUT)
 		REMOVE_TRAIT(src, TRAIT_TOOL_WRENCH, TRAIT_SOURCE_INHERENT)
 		ADD_TRAIT(src, TRAIT_TOOL_CROWBAR, TRAIT_SOURCE_INHERENT)
+		pry_capable = IS_PRY_CAPABLE_FORCE
+	. = ..()
```
Comment (harryob) [inline code/game/objects/items/tools/maintenance_tools.dm:538]: can we do . = ..() at the start of the proc and return after the if(crowbar_mode) and removing the else block

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**